apps/events-helsinki/src/__tests__/redirectRoutes.test.tsx

```diff
@@ -0,0 +1,68 @@
+import assert from 'assert';
+import type { IncomingMessage } from 'http';
+import http from 'http';
+import Link from 'next/link';
+import router from 'next/router';
+import { render, waitFor, screen, userEvent } from '@/test-utils';
+import AppConfig from '../../src/domain/app/AppConfig';
+
+const APP_DOMAIN = new URL(AppConfig.origin).origin;
+
+// Skip until there is a fix for this NextJS bug https://github.com/vercel/next.js/discussions/26426
+describe.skip('NextJS redirects', () => {
+  it.each([
+    ['/fi/home', '/fi'],
+    ['/sv/home', '/sv'],
+    ['/en/home', '/en'],
+    ['/fi/events', '/fi/haku'],
+    ['/sv/events', '/sv/sok'],
+    ['/en/events', '/en/search'],
+  ])(
+    'should redirect the requests of the old app-version path (%s) to the new path (%s)',
+    async (redirectFrom, redirectTo) => {
+      render(
+        <Link href={redirectFrom}>
+          <a>
+            from {redirectFrom} to {redirectTo}
+          </a>
+        </Link>
+      );
+      const link = screen.getByRole('link');
+      userEvent.click(link);
+      await waitFor(() => {
+        expect(router.asPath).toBe(redirectTo);
+      });
+    }
+  );
+});
+
+describe('Test permanent redirects', () => {
+  it.each([
+    ['/fi/home', '/fi'],
+    ['/sv/home', '/sv'],
+    ['/en/home', '/en'],
+    ['/fi/events', '/fi/haku'],
+    ['/sv/events', '/sv/sok'],
+    ['/en/events', '/en/search'],
+    ['/fi/events?categories=food%2Cdance', '/fi/haku?categories=food%2Cdance'],
+    ['/sv/events?categories=food%2Cdance', '/sv/sok?categories=food%2Cdance'],
+    [
+      '/en/events?categories=food%2Cdance',
+      '/en/search?categories=food%2Cdance',
+    ],
+  ])(
+    'redirects %s to %s with permanent redirect',
+    async (redirectFrom, redirectTo) => {
+      const incomingMessage: IncomingMessage = await new Promise((resolve) => {
+        assert(!APP_DOMAIN.endsWith('/'));
+        assert(redirectFrom.startsWith('/'));
+        http.get(`${APP_DOMAIN}${redirectFrom}`, (res) => {
+          resolve(res);
+        });
+      });
+
+      expect(incomingMessage.statusCode).toBe(308); // 308 = Permanent Redirect
+      expect(incomingMessage.headers.location).toBe(redirectTo);
+    }
+  );
+});
```

Comment:
Where does this test mount a NextJS with the new redirect configuration?

Reply:
It doesn't, I probably had the next.js server running from apps/events-helsinki/ using "yarn dev" when I was running these tests locally. I think it's not that much worth the while to put much more time into this PR, I removed the http.get using jest test for redirects that weren't working without the next.js server having been loaded beforehand.